cdap-app-fabric/src/main/java/co/cask/cdap/internal/app/runtime/schedule/StreamSizeScheduler.java

```diff
@@ -138,4 +316,470 @@ private String getProgramScheduleId(Id.Program program, SchedulableProgramType p
+      }
+
+      if (lastNotification != null) {
+        // In all cases, when creating a schedule - either if it comes from the store or not,
+        // we want to pass it the last seen notification if it exists.
+        // This call will send the notification to all the active tasks. The ones which are active before
+        // that method is called will therefore see the notification twice. It is fine though.
+        received(lastNotification, null);
+      }
+      return true;
+    }
+
+    /**
+     * Suspend a scheduling task that is based on the data received by the stream referenced by {@code this} object.
+     */
+    public synchronized void suspendScheduleTask(Id.Program programId, SchedulableProgramType programType,
+                                                 String scheduleName) throws ScheduleNotFoundException {
+      String scheduleId = getScheduleId(programId, programType, scheduleName);
+      StreamSizeScheduleTask task = scheduleTasks.get(scheduleId);
+      if (task == null) {
+        throw new ScheduleNotFoundException(scheduleName);
+      }
+      if (task.suspend()) {
+        activeTasks.decrementAndGet();
+      }
+    }
+
+    /**
+     * Resume a scheduling task that is based on the data received by the stream referenced by {@code this} object.
+     */
+    public void resumeScheduleTask(Id.Program programId, SchedulableProgramType programType, String scheduleName)
+      throws ScheduleNotFoundException {
+      int activeTasksNow;
+      StreamSizeScheduleTask task;
+      synchronized (this) {
+        String scheduleId = getScheduleId(programId, programType, scheduleName);
+        task = scheduleTasks.get(scheduleId);
+        if (task == null) {
+          throw new ScheduleNotFoundException(scheduleName);
+        }
+        if (!task.resume()) {
+          return;
+        }
+        activeTasksNow = activeTasks.incrementAndGet();
+      }
+      if (activeTasksNow == 1) {
+        // There were no active tasks until then, that means polling the stream was disabled.
+        // We need to check if it is necessary to poll the stream at this time, if the last
+        // notification received was too long ago, or if there is no last seen notification
+        synchronized (lastNotificationLock) {
+          if (lastNotification == null ||
+            (lastNotification.getTimestamp() + pollingDelay <= System.currentTimeMillis())) {
+            // Resume stream polling
+            cancelPollingAndScheduleNext();
+            try {
+              StreamSize streamSize = getStreamEventsSize();
+              lastNotification = new StreamSizeNotification(streamSize.getTimestamp(), streamSize.getSize());
+            } catch (IOException e) {
+              LOG.debug("Ignoring stream events size polling after resuming schedule {} due to error", scheduleName, e);
+            }
+          }
+        }
+      }
+      task.received(lastNotification);
```

Comment:
I think that's what we want here, only one task re-activates the polling. This task will recompute a notification using the polled size, and the second task will be sent that notification too (because ``task.received`` is outside of the ``if (activeTasksNow == 1)`` block).

There could one problem though, if we have the following:
- Task1 resumes, sees activeTasksNow == 1
- Task2 resumes, sees activeTasksNow == 2
- Task2 doesn't go into condition, and executes ``task.received`` with the oldest ``lastNotification``
- Task1 enters the condition, and recomputes ``lastNotification``
- Task1 executes ``task.received`` with the new ``lastNotification``

So depending on when things execute, Task2 may not benefit from the new computed ``lastNotification`` that Task1 is supposed to provide. Again, I thought I could avoid synchronizing the method completely, but it doesn't look like it...